core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -97,7 +97,7 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDir) {
+    if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDirectory) {
```

Comment:
Unrelated to this line: this class has the `hadoopFlushMethod` hack which probably can go away now, if you want to do more cleanup.